scripts/list_shields.py

```diff
@@ -66,9 +68,19 @@ def add_args(parser):
                         type=Path, action='append',
                         help='add a board root, may be given more than once')
 
+def add_args_formatting(parser):
+    parser.add_argument("--json", action='store_true',
+                        help='''output list of shields in JSON format''')
+
 def dump_shields(shields):
-    for shield in shields:
-        print(f'  {shield.name}')
+    if args.json:
+        print(
+            json.dumps([{'dir': shield.dir.as_posix(), 'name': shield.name} for shield in shields])
```

Comment:
I guess that, in the future, you would like to list the snippets in the same way. The long-term plan for snippets support is to improve it to the level of the shields and make shields a subset of snippets. Snippets (scripts/snippet.py) dumps the snippets list into snippets_generated.cmake. Not sure if using JSON is faster @tejlmand ?